solver/llbsolver/ops/source.go

```diff
@@ -73,15 +73,20 @@ func (s *sourceOp) CacheMap(ctx context.Context, g session.Group, index int) (*s
 	}
 
 	dgst := digest.FromBytes([]byte(sourceCacheType + ":" + k))
-
 	if strings.HasPrefix(k, "session:") {
 		dgst = digest.Digest("random:" + strings.TrimPrefix(dgst.String(), dgst.Algorithm().String()+":"))
 	}
 
+	var resolveResponse map[string]string
+	if !strings.HasPrefix(s.op.Source.GetIdentifier(), "local://") {
+		resolveResponse = map[string]string{s.op.Source.GetIdentifier(): k}
```

Comment:
edit: (left wrong comment before, ignore that if you got notified).

We need to verify that these are the correct hashes for the object. It is not quite the same as the internal cache key in some cases. Eg. https://github.com/moby/buildkit/blob/b2195cec759428a7775a6fe01799b961d14a81bc/source/containerimage/pull.go#L155 adds more info to cache key to make sure platform is correct(+ there is another config based key as well that we should ignore). https://github.com/moby/buildkit/blob/b2195cec759428a7775a6fe01799b961d14a81bc/source/git/gitsource.go#L173-L178 does same for git. In here we would need the raw root image digest, git commit sha, raw sha256 of URL etc.